queued_sale_notify/models/sale_order.py

```diff
@@ -16,3 +22,24 @@ def action_confirm(self):
                     if notify_id.is_to_notify(rec):
                         notify_id.notify(rec)
         return res
+
+    def _compute_mail_queues(self):
```

Comment:
You can use this method to compute both mail_queue_ids and  has_queues field. Moreover, I don't think that the has_queues field is necessary, as it is simply used to show the queues page in a sale order view, which can be done by checking if mail_queue_ids has values.